lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -508,8 +510,17 @@ def response(action, succeeded, response, options = {}, source_id = nil)
       end
 
       def headers(action, options)
-        auth_token = @access_token ? "Bearer #{@access_token}" : @options[:secret_key]
-        auth_token = @options[:public_key] if action == :tokens
+        auth_token =
```

Comment:
I would prefer if these was it's own PR like it was before. It would make it a lot easier to diagnose a problem if something goes wrong when it's deployed.

Reply:
Sure, I will

Reply:
Please don't forget to make this into it's on PR

Reply:
I did it, it's just that in this same PR I added the changes from the ticket: ECS-3487 about Update Authorization from Basic to Bearer for Checkout gateway

Reply:
I know but it's better if these are in two separate PRs. One for retaining and refreshing the access_token and another for updating the authorization. Keeping this two changes separate will make it easier to diagnose if something goes wrong when we merge it.

Reply:
ok, I will

Reply:
I removed these changes from this PR, they caused a lot of confusion and they will be in a new PR